docs/src/main/sphinx/connector/object-storage-cache.md

```diff
@@ -0,0 +1,114 @@
+accessing the object storage is low compared to accessing the cache.
+
+For example, if you access object storage in a different network, different data
+center, or even different cloud-provider region query performance is slow. Adding
+caching using fast, local storage has a significant impact and makes your
+queries much faster.
+
+On the other hand, if your object storage is already running at very high
+performance for I/O and network access, and your local cache storage is at
+similar speeds, or even slower, performance benefits can be minimal.
+
+**Reduced query costs**
+
+A result of the reduced load on the object storage, mentioned earlier, is
+significantly reduced network traffic. Network traffic is often a considerable
+cost factor, specifically also when hosted in public cloud provider systems.
+
+## Configuration
+
+Use the properties from the following table in your catalog properties files to
+enable and configure caching for the specific catalogs.
+
+:::{list-table} Cache configuration properties
+:widths: 25, 75
+:header-rows: 1
+
+* - Property
+  - Description
+* - `fs.cache`
+  - Configure the caching type. Defaults to no caching with the value `none`.
+    Set to `alluxio` to enable object storage caching powered by Alluxio.
+* - `fs.cache.directories`
+  - Required, comma-separated list of absolute paths to directories to use for
+    caching. All directories must exist on the coordinator and all workers.
+    Trino must have read and write permissions for files and nested directories.
+    A valid example with only one directory is `/tmp/trino-cache`.
+* - `fs.cache.max-sizes`
+  - Optional, comma-separated list of maximum [data size](prop-type-data-size) for
+    the each caching directory. Order of values must be identical to the
+    directories list. Can not be used together with
+    `fs.cache.max-disk-usage-percentages`.
+* - `fs.cache.max-disk-usage-percentages`
+  - Optional, comma-separated list of maximum percentage values of the used disk
+    for each directory. Each value is an integer between 1 and 100. Order of
+    values must be identical to the directories list. If multiple directories
+    use the same disk, ensure that total percentages per drive remains below 100
+    percent. Can not be used together with `fs.cache.max-sizes`.
+* - `fs.cache.ttl`
+  -  The maximum [duration](prop-type-duration) for objects to remain in the cache
+     before eviction. Defaults to `7d`. The minimum value of `0s` means that caching
+     is effectively turned off.
+* - `fs.cache.preferred-hosts-count`
+  - The preferred number of nodes for caching each file. Defaults to 2. If a
+    file is cached on a node and required by another node, the file is retrieved
+    from that node rather than object storage. Defaults to 2. Values higher than
+    the number of nodes in the cluster results in caching files potentially on
+    every node as needed by query processing.
```

Comment:
There's no mechanism for inter node cached reads in alluxio. The file is always retrieved from object storage if it's not found locally.
This config controls the number of hosts that will be identified for caching and presented to the scheduler. The scheduler is going to try these nodes first and then it's free to schedule the split anywhere if those nodes are busy or removed from the cluster. Ideally we should have code to prevent caching on a worker if it's not one of the "preferred" nodes, but we don't have that implemented in the current PR.

Reply:
Updated anyway..